src/ucp/wireup/address.c

```diff
@@ -991,6 +1000,10 @@ ucs_status_t ucp_address_unpack(ucp_worker_t *worker, const void *buffer,
     }
 
     if (unpack_flags & UCP_ADDRESS_PACK_FLAG_WORKER_UUID) {
+        ucs_assert(address_header & UCP_ADDRESS_HEADER_FLAG_WORKER_UUID);
+    }
+
```

Comment:
i'd combine these 2 `if` branches to a single one

Reply:
Second if doesn't consider unpack flag. We now ignore unpack flag if the address was packed with WORKER_UUID we need to unpack it no matter if the user passed unpack flag or not. Just to double-check if the user passed a flag WORKER_UUID to make sure it's presented in header flags.